irohad/ametsuchi/impl/rocksdb_common.hpp

```diff
@@ -853,15 +857,15 @@ namespace iroha::ametsuchi {
       status = common.put(std::forward<Args>(args)...);
     else if constexpr (kOp == kDbOperation::kDel)
       status = common.del(std::forward<Args>(args)...);
-
+    
     static_assert(kOp == kDbOperation::kGet || kOp == kDbOperation::kCheck
                       || kOp == kDbOperation::kPut || kOp == kDbOperation::kDel,
                   "Unexpected operation value!");
 
     static_assert(
-        kOp != kDbOperation::kDel || kSc != kDbEntry::kMustExist,
+        kOp != kDbOperation::kDel || kSc == kDbEntry::kCanExist,
```

Comment:
The behaviour the same as std::map(for ex.) When you try to delete unexisted key it will correctly compile and work. It doesn't drop you compilation or throw even if ```key``` is not exists.